src/oscar/apps/partner/abstract_models.py

```diff
@@ -232,6 +232,8 @@ def consume_allocation(self, quantity):
         This is used when an item is shipped.  We remove the original
         allocation and adjust the number in stock accordingly
         """
+        if not self.product.product_class.track_stock:
```

Comment:
Need to apply the same logic in `StockRecord.allocate()`. It doesn't make sense to allocate stock if it isn't being tracked.